infrastructure/code/cache.go

```diff
@@ -0,0 +1,109 @@
+/*
+ * © 2024 Snyk Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package code
+
+import (
+	"github.com/erni27/imcache"
+
+	"github.com/snyk/snyk-ls/domain/snyk"
+	"github.com/snyk/snyk-ls/internal/product"
+)
+
+func (sc *Scanner) addToCache(results []snyk.Issue) {
```

Comment:
This question doesn't relate to this PR necessarily but I'm wondering why we need to have these functions in the code package? There doesn't seem to be anything code specific in here and I'm wondering if these functions can be re-used by other scanners to. So in the delegating concurrent scanner instead of delegating down for this for no reason we do all this in there

Reply:
Nevermind, ignore me. `sc.issueCache` is Code's cache

Reply:
Yeah, most likely it could be extracted to be used by all scanners. That may be a good refactor when we update the other scanners to have their own caches instead of using a global cache.